.github/workflows/ci-post-land.yml

```diff
@@ -329,31 +329,31 @@ jobs:
 
   rustdoc:
     needs: prepare
-    if: ${{ needs.prepare.output.changes-target-branch == 'main' }}
-    env:
-      CARGO_INCREMENTAL: 0
-      RUSTFLAGS: -D warnings
+    if: ${{ needs.prepare.outputs.changes-target-branch == 'main' && needs.prepare.outputs.rust-changes == 'true'}}
```

Comment:
Triggering on the right input "output" -> "outputs", kept this from firing before.    Also, don't rerun if no rust changes were found.